pkg/trace/agent/agent.go

```diff
@@ -605,6 +612,48 @@ func isManualUserDrop(pt *traceutil.ProcessedTrace) bool {
 	return dm == manualSampling
 }
 
+// errorSampling reports trace chunks with errors and tags spans as Error Tracking Backend Standalone.
+// Also sets "DroppedTrace" on the chunk.
+func (a *Agent) errorSampling(now time.Time, ts *info.TagStats, pt *traceutil.ProcessedTrace) (keep bool, numEvents int) {
+	sampled := a.runErrorSampler(now, *pt)
+	numEvents = len(a.getAnalyzedEvents(pt, ts))
+	if sampled {
+		for _, span := range pt.TraceChunk.Spans {
+			if span.Error != 0 || spanContainsExceptionSpanEvent(span) {
+				span.Meta["_dd.error_tracking_backend_standalone.error"] = "true"
+			} else {
+				span.Meta["_dd.error_tracking_backend_standalone.error"] = "false"
+			}
+		}
+	}
+	pt.TraceChunk.DroppedTrace = !sampled
+	return sampled, numEvents
+}
+
+func spanContainsExceptionSpanEvent(span *pb.Span) bool {
+	if hasExceptionSpanEvents, ok := span.Meta["_dd.span_events.has_exception"]; ok && hasExceptionSpanEvents == "true" {
+		return true
+	}
+	return false
+}
+
+// runErrorSampler runs the agent's configured ErrorSampler if pt contains errors and returns the sampling decision.
+func (a *Agent) runErrorSampler(now time.Time, pt traceutil.ProcessedTrace) (keep bool) {
+	if traceContainsErrorOrExceptionSpanEvent(pt.TraceChunk.Spans) {
+		return a.ErrorsSampler.Sample(now, pt.TraceChunk.Spans, pt.Root, pt.TracerEnv)
+	}
+	return false
+}
+
+func traceContainsErrorOrExceptionSpanEvent(trace pb.Trace) bool {
```

Comment:
We have a very similar function already, maybe it's worth extending `traceContainsError()` adding a boolean parameter to determine whether exceptions should be considered?

Reply:
that also works! done in https://github.com/DataDog/datadog-agent/pull/30065/commits/ce0d656b2e5348fe6bb7b3afd91c9cce238b89e0